key/key.go

```diff
@@ -1,108 +1,209 @@
 package key
 
 import (
+	"bytes"
 	"encoding/hex"
 	"fmt"
 	"math"
-	"strings"
-)
 
-type KadKey []byte
+	"github.com/plprobelab/go-kademlia/kad"
+)
 
-func (k KadKey) Size() int {
-	return len(k)
+// Key256 is a 256-bit Kademlia key.
+type Key256 struct {
+	bits *[32]byte // this is a pointer to keep the size of Key256 small since it is often passed as argument
```

Comment:
It's actually `bytes` and not `bits`. This confused me in the for loop in L57. We're not indexing individual bits but bytes.

`bytes` collides with the go package, though. Might be okay because it's a field on a struct.

Reply:
should be fine to change

Reply:
changed to `b` to avoid confusion